File: tests/fixtures/greeter.py
"""Simple greeter module."""


def greet(name: str) -> str:
    """Return a greeting."""
    return f"Hello, {name}"


def farewell(name: str) -> str:
    """Return a farewell."""
    return f"Goodbye, {name}"


class Greeter:
    """A greeter class."""

    def __init__(self, greeting: str = "Hello"):
        self.greeting = greeting

    def greet(self, name: str) -> str:
        """Greet someone by name."""
        return f"{self.greeting}, {name}!"

File: tests/fixtures/greeter.ts
/** Simple greeter module. */

export interface GreeterOptions {
  greeting: string;
}

export class Greeter {
  private greeting: string;

  constructor(options: GreeterOptions) {
    this.greeting = options.greeting;
  }

  public greet(name: string): string {
    return `${this.greeting}, ${name}!`;
  }
}

export function freeFunction(name: string): string {
  return `Hello, ${name}`;
}
